File: Patterns/python/pyramidOfNumber.py
# n = 4

#    1
#   212
#  32123
# 4321234

n = int(input())
row = 1
while row <= n:
    space = 1
    while space <= n - row:
        print(" ", end="")
        space += 1
    
    col = 1
    val = row
    while col <= row:
        print(val, end="")
        val -= 1
        col += 1
    
    col = 1
    val += 2
    while col <= row - 1:
        print(val, end="")
        col += 1
        val += 1
    
    row += 1
    print("")

File: Patterns/js/pyramidOfNumber.js
/*
n = 4

   1
  212
 32123
4321234

*/

function pattern(n) {
  let ans = "";
  let row = 1;

  while (row <= n) {
    let space = 1;
    while (space <= n - row) {
      ans += " ";
      space++;
    }

    let col = 1;
    let val = row;
    while (col <= row) {
      ans += val;
      val--;
      col++;
    }

    col = 1;
    val += 2;
    while (col <= row - 1) {
      ans += val;
      val++;
      col++;
    }

    row++;
    ans += "\n";
  }

  console.log(ans);
}

pattern(4);
